sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/udf/CosmosAsyncClientCache.scala

```diff
@@ -20,13 +20,13 @@ object CosmosAsyncClientCache {
    */
   def getCosmosClientFromCache(userProvidedConfig: Map[String, String]): CosmosAsyncClientCacheItem = {
     val effectiveUserConfig = CosmosConfig.getEffectiveConfig(None, None, userProvidedConfig)
-    val cosmosClientConfig = CosmosClientConfiguration(
-      effectiveUserConfig,
-      useEventualConsistency = false,
-      CosmosClientConfiguration.getSparkEnvironmentInfo(SparkSession.getActiveSession))
+    val readConfig = CosmosReadConfig.parseCosmosReadConfig(effectiveUserConfig)
+    val useEventualConsistency = readConfig.forceEventualConsistency
+    val sparkEnvironmentInfo = CosmosClientConfiguration.getSparkEnvironmentInfo(SparkSession.getActiveSession)
 
-    new CosmosAsyncClientCacheItem(
-      CosmosClientCache(cosmosClientConfig, None, "CosmosAsyncClientCache.getCosmosClientFromCache"))
+    // delay getting the client from cache here to allow this to be executed on executors
```

Comment:
nit: remove the comment as in this method the client creation will not be delayed?

Reply:
done